util/helm/cmd.go

```diff
@@ -0,0 +1,165 @@
+package helm
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"regexp"
+
+	argoexec "github.com/argoproj/pkg/exec"
+)
+
+// A thin wrapper around the "helm" command, adding logging and error translation.
+type Cmd struct {
+	helmHome string
+	WorkDir  string
+}
+
+func NewCmd(workDir string) (*Cmd, error) {
+	tmpDir, err := ioutil.TempDir("", "helm")
+	if err != nil {
+		return nil, err
+	}
+	return &Cmd{WorkDir: workDir, helmHome: tmpDir}, err
+}
+
+var redactor = func(text string) string {
+	return regexp.MustCompile("(--username|--password) [^ ]*").ReplaceAllString(text, "$1 ******")
```

Comment:
I think I have seen a change lately that redacts other confidential information to character `+` instead of `*`, but I'm unsure. Gotta have to find it.

Reply:
i know the PR, and good spot, this specific case we are ok